### PR TITLE
Create and publish SDK docs in gh-pages

### LINKED ISSUE #94: Rust Docs
Research around the rust docs, and fix the configuration, in order to be able to generate docs without the dependencies (and maybe publish them to github-pages as well?).

### PR DESCRIPTION
**Pull request by wirednkod**
Fixes #94

### CODE REVIEW

**Review by pepoviola — APPROVED** [review]

**Comment by l0r1s** [comment]
Hey, nice job, do we really need to commit all assets including build files ?

**Comment by pepoviola** [comment]
> Hey, nice job, do we really need to commit all assets including build files ?

That a good question, should be a way to publish those without having them committed (at least not in main branch).

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Cargo Create Docs
+
+on:
+  push:
+    branches: [nik-add-docs-in-CI]
```
Comment (l0r1s) [inline .github/workflows/documentation.yml:5]: Just modify this to "main" and should be good for merge :+1:

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  pull_request:
+    branches: [nik-add-docs-in-CI]
```
Comment (l0r1s) [inline .github/workflows/documentation.yml:7]: Same here

**Review by l0r1s — APPROVED** [review]
lgtme ! just changing the branches and ready to merge :)

File: .github/workflows/documentation.yml (hunk 1 above)
Comment (wirednkod) [reply .github/workflows/documentation.yml:5]: still testing.. this is not working

File: .github/workflows/documentation.yml (hunk 2 above)
Comment (wirednkod) [reply .github/workflows/documentation.yml:7]: still testing.. this is not working

**Pull request merged**